viam-cartographer.go

```diff
@@ -153,28 +156,57 @@ func New(
 		}
 	}()
 
-	if cartoSvc.primarySensorName != "" {
-		if err := dim2d.ValidateGetAndSaveData(cancelCtx, cartoSvc.dataDirectory, lidar,
-			sensorValidationMaxTimeoutSec, sensorValidationIntervalSec, cartoSvc.logger); err != nil {
-			return nil, errors.Wrap(err, "getting and saving data failed")
+	if modularizationV2Enabled {
+		// Initialize the library
+		// TODO: get these correctlty
+		loglevel := 1
+		verbosity := 1
+		cartofacade.Lib, err = cartofacade.NewLib(loglevel, verbosity)
+		if err != nil {
+			return nil, err
 		}
-		cartoSvc.StartDataProcess(cancelCtx, lidar, nil)
-		logger.Debugf("Reading data from sensor: %v", cartoSvc.primarySensorName)
-	}
 
-	if err := cartoSvc.StartSLAMProcess(ctx); err != nil {
-		return nil, errors.Wrap(err, "error with slam service slam process")
-	}
+		// Get the config
+		cartoCfg := cartofacade.CartoConfig{
+			Sensors:            svcConfig.Sensors,
+			MapRateSecond:      dataRateMsec * 1000,
+			DataDir:            svcConfig.DataDirectory,
+			ComponentReference: svcConfig.Sensors[0],
```

Comment:
@nicksanford can `ComponentReference` be removed now with your changes to use `sensors[0]` in place of every where it was used, or is it needed for something else?